src/feedClass.ts

```diff
@@ -0,0 +1,281 @@
+        logger.info('class login');
+        const user = await this.gunAuthInitFeed(username, password);
+        return (user)
+    }
+
+    /**
+     * add item to feed
+     * @param {string} feedInfoName
+     * @param {FeedItem} feedItem
+     */
+    async addItem(feedInfoName: string, feedItem: FeedItem): Promise<boolean> {
+        const date = new Date();
+        const day = `${date.toISOString().slice(0, 10)}`;
+
+        await this.feeds.get(feedInfoName).get('items').get(day).put(feedItem);
+        return true
+    }
+
+    /**
+     * Add private item to feed
+     * @param {string} feedInfoName
+     * @param {FeedItem} feedItem
+     * @param {string} privateKey
+     */
+    async addPrivateItem(feedInfoName: string, feedItem: FeedItem, privateKey: string): Promise<boolean> {
+        const date = new Date();
+        const day = `${date.toISOString().slice(0, 10)}`;
+
+        await this.feeds.get(feedInfoName).get('privateItems').get(privateKey).get(day).put(feedItem);
+        return true
+    }
+
+    /**
+     * Create new feed
+     *
+     * @param {Feed} feed
+     */
+    async createFeed(feed: Feed): Promise<Feed | any> {
+        logger.info('class createFeed');
+
+        feed.info.owner = await this.getOwnerSoul();
+        const gunNewFeed = this.feeds.get(feed.info.name).put(feed);
+        const id = await this.getSoul(gunNewFeed);
+        await gunNewFeed.get('info').get('id').put(id);
+        return gunNewFeed;
+    }
+
+
+    /**
+     * Return user soul
+     */
+    async getOwnerSoul(): Promise<string> {
+        return await this.getSoul(this.gunUser)
+    }
+
+
+    /**
+     * Return soul gun object
+     * @param {Gun} gunInstance
+     */
+    async getSoul(gunInstance: Gun): Promise<string> {
+        return (await gunInstance)['_']['#']
+    }
+
+    /**
+     * Get feed by id
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeed(feedInfoName: string): Promise<Feed | any> {
+        logger.info('class getFeed', {feedInfoName});
+        let result: any = {
+            info: {},
+            items: {},
+            privateItems: {}
+        };
+
+        try {
+            const info = await this.feeds.get(feedInfoName).get('info').then()
+
+            if (info) {
+                result.info = _.pick(info, this.feedInfoFields);
+                result.items = await this.getFeedItems(feedInfoName);
+                result.privateItems = await this.getFeedPrivateItems(feedInfoName)
+            }
+
+        } catch (e) {
+            logger.error(e);
+        }
+
+        return result
+    }
+
+    /**
+     * Return array items
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeedItems(feedInfoName: string): Promise<FeedItem | any> {
+        let result: any = {};
+        logger.info('class getFeedItems', {feedInfoName});
+        const items = await this.feeds.get(feedInfoName).get('items').then();
```

Comment:
- keyItem's ref could be stored then re-used.
- .map isn't needed as we don't have any pass-throudg call.
Also, .map() is still in experimental mode and isn't recommended for production usage
- better to use Promise.all (run paralelly)
- we can use zipObject from lodash

```
import { pick, zipObject } from 'lodash';
import gunDb, { fetchItemsRef } from './lib/gundb';

class FeedsCollection {
  async getFeedItems(feedInfoName: string): Promise<FeedItem> {
    const { feedsRef, feedItemFields, feedItemDataFields } = this;
    const feedItemsRef = feedsRef.path(`${feedInfoName}.items`);
    const itemsKeys = await fetchItemsRef(feedItemsRef);

    const itemsData = await Promise.all(itemsKeys.map(async itemKey => {
      const itemRef = feedItemsRef.get(itemKey);
      const mainFields = await itemRef.then();
      const dataFields = await itemRef.get('data').then();

      return ({
        ...pick(mainFields, feedItemFields),
        data: pick(dataFields, feedItemDataFields)
      });
    }));

    return zipObject(itemsKeys, itemsData);
  }
}
```